### PR TITLE
[GH-15535] Python 3.11 support

### LINKED ISSUE #15535: Support for Python 3.10 and 3.11 in H2O Packages
The following H2O packages are currently unavailable for Python 3.10 in client's Artifactory:

JIRA ticket: https://h2oai.atlassian.net/browse/PUBDEV-9099

h2o
h2o_pysparkling_3.0
h2o_pysparkling_scoring_3.0

For more context, support ticket#: 105447

### PR DESCRIPTION
**Pull request by mn-mikke**
closes #15535

As part of this PR, there are new testing images:

harbor.h2o.ai/opsh2oai/h2o-3/dev-python-3.11:42
harbor.h2o.ai/opsh2oai/h2o-3/dev-python-3.11-jdk-8:42
harbor.h2o.ai/opsh2oai/h2o-3/dev-python-3.11-jdk-11:42
harbor.h2o.ai/opsh2oai/h2o-3/dev-python-3.11-jdk-17:42

This pr also moves al python small stages to PR tests (PR test are also a part of nightly).

### CODE REVIEW

File: scripts/jenkins/groovy/defineTestStages.groovy [hunk 1]
```diff
@@ -45,6 +45,10 @@ def call(final pipelineContext) {
       stageName: 'Py3.10 Smoke', target: 'test-py-smoke', pythonVersion: '3.10', timeoutValue: 8,
       component: pipelineContext.getBuildConfig().COMPONENT_PY
     ],
+    [
+      stageName: 'Py3.11 Smoke', target: 'test-py-smoke', pythonVersion: '3.11', timeoutValue: 8,
```
Comment (valenad1) [inline scripts/jenkins/groovy/defineTestStages.groovy:49]: I think smoke stages was only for Lowest and Highest version.

File: scripts/jenkins/groovy/defineTestStages.groovy [hunk 2]
```diff
@@ -508,10 +520,6 @@ def call(final pipelineContext) {
       stageName: 'Py3.6 Single Node', target: 'test-pyunit-single-node', pythonVersion: '3.6',
       timeoutValue: 40, component: pipelineContext.getBuildConfig().COMPONENT_PY
     ],
-    [
-      stageName: 'Py3.6 Small', target: 'test-pyunit-small', pythonVersion: '3.6',
```
Comment (valenad1) [inline scripts/jenkins/groovy/defineTestStages.groovy:512]: Why is this removed?

Comment (valenad1) [reply scripts/jenkins/groovy/defineTestStages.groovy:512]: its moved, sorry, I see it now

File: scripts/jenkins/groovy/defineTestStages.groovy [hunk 3]
```diff
@@ -529,23 +537,19 @@ def call(final pipelineContext) {
       timeoutValue: 90, component: pipelineContext.getBuildConfig().COMPONENT_PY
     ],
     [
-      stageName: 'Py3.10 Single Node', target: 'test-pyunit-single-node', pythonVersion: '3.10',
```
Comment (valenad1) [inline scripts/jenkins/groovy/defineTestStages.groovy:532]: These tests are also only for oldest and latest java?

**Review by valenad1 — DISMISSED** [review]
Couple of questions ❓

Looks good to me, thank you!

File: scripts/jenkins/groovy/defineTestStages.groovy (hunk 1 above)
Comment (mn-mikke) [reply scripts/jenkins/groovy/defineTestStages.groovy:49]: Yes, good catch! I forgot about to remove python 3.10

File: scripts/jenkins/groovy/defineTestStages.groovy (hunk 3 above)
Comment (mn-mikke) [reply scripts/jenkins/groovy/defineTestStages.groovy:532]: yes, oldest and latest supported python

File: h2o-py/test-requirements.txt [hunk 4]
```diff
@@ -35,7 +36,7 @@ lifelines>=0.19.5
 xlsxwriter==1.3.7
 xgboost==1.7.6
 causalml==0.12.3; python_version <= '3.7'
-causalml==0.14.1; python_version > '3.7'
+causalml==0.14.1; python_version > '3.7' and python_version < '3.11'
```
Comment (wendycwong) [inline h2o-py/test-requirements.txt:39]: should this be:

and python_version <= '3.11'

**Review by wendycwong — APPROVED** [review]

File: h2o-py/test-requirements.txt (hunk 4 above)
Comment (mn-mikke) [reply h2o-py/test-requirements.txt:39]: Nope, causalml doesn't support python 3.11 yet.
